### PR TITLE
[SYCL] Add support multidimensional subscript for atomic accessor

### PR DESCRIPTION
**Pull request by maximdimakov**
Add support for multidimensional subscript operations for atomic accessor
The test : https://github.com/intel/llvm-test-suite/pull/218
Signed-off-by: mdimakov <maxim.dimakov@intel.com>

### CODE REVIEW

File: sycl/include/CL/sycl/accessor.hpp [hunk 1]
```diff
@@ -325,6 +325,14 @@ class accessor_common {
       return MAccessor[MIDs];
     }
 
+    template <int CurDims = SubDims>
+    typename detail::enable_if_t<
+        CurDims == 1 && AccessMode == access::mode::atomic, atomic<DataT, AS>>
+    operator[](size_t Index) const {
+      MIDs[Dims - CurDims] = Index;
+      return MAccessor[MIDs];
```
Comment (dm-vodopyanov) [inline sycl/include/CL/sycl/accessor.hpp:333]: should it be some explicit construction of an object of `atomic<DataT, AS>` type?

Comment (maximdimakov) [reply sycl/include/CL/sycl/accessor.hpp:333]: The sycl::atomic is constructed in the accessor class, when we call MAccessor[MIDs]. We do not create any objects in AccessorSubscript::operator[] methods, only call operator[] for the accessor. I think it is more correct to create sycl::atomic out of this class.

File: sycl/include/CL/sycl/accessor.hpp [hunk 2]
```diff
@@ -325,6 +325,14 @@ class accessor_common {
       return MAccessor[MIDs];
     }
 
+    template <int CurDims = SubDims>
+    typename detail::enable_if_t<
+        CurDims == 1 && AccessMode == access::mode::atomic, atomic<DataT, AS>>
```
[the remark this answers is not in the document]
Comment (maximdimakov) [reply sycl/include/CL/sycl/accessor.hpp]: Added

**Comment by s-kanaev** [comment]
@maximdimakov , please, add tests.
I'm not sure whether the tests should be added to intel/llvm or intel/llvm-test-suite repo.

File: sycl/include/CL/sycl/accessor.hpp (hunk 1 above)
Comment (dm-vodopyanov) [reply sycl/include/CL/sycl/accessor.hpp:335]: Sounds good, thanks

**Review by dm-vodopyanov — APPROVED** [review]

**Pull request merged**